morango/models/fields/uuids.py

```diff
@@ -2,67 +2,35 @@
 import uuid
 
 from django.db import models
+
 from morango.utils import _assert
 
 
 def sha2_uuid(*args):
     return hashlib.sha256("::".join(args).encode("utf-8")).hexdigest()[:32]
 
 
-class UUIDField(models.CharField):
+class UUIDField(models.UUIDField):
     """
     Adaptation of Django's UUIDField, but with 32-char hex representation as Python representation rather than a UUID instance.
     """
 
-    def __init__(self, *args, **kwargs):
-        kwargs["max_length"] = 32
-        super(UUIDField, self).__init__(*args, **kwargs)
-
-    def prepare_value(self, value):
-        if isinstance(value, uuid.UUID):
-            return value.hex
-        return value
```

Comment:
This one doesn't seem to be captured in the parent class.

Reply:
(and is an important part of having it return the 32 char value rather than a UUID object, from what I recall)

Reply:
(I think maybe specifically for cases where someone sets it on there as a UUID object and then reads it back, without it going through the database yet)

But it could probably call `to_python` here to be DRYer.

Reply:
`prepare_value` is a form field method though, not a regular field: https://github.com/django/django/blob/stable/1.11.x/django/forms/fields.py#L129

I do wonder if that's where some of the original implementation errors came from, as it seems the original implementation here was a copy of the Django form field UUIDField: https://github.com/django/django/blob/stable/1.11.x/django/forms/fields.py#L1206 but then wrapped around the model CharField implementation instead of the form field.

Reply:
There is definitely something missing though, because during the serialization tests, it is returning as a UUID.

Reply:
The default Django serialization framework uses the `value_to_string` method, which we seem to have not defined here.

See: https://github.com/django/django/blob/stable/1.11.x/django/db/models/fields/__init__.py#L834

Although we're then not using that in the serialize method: https://github.com/learningequality/morango/blob/release-v0.6.x/morango/models/core.py#L939

Reply:
Have now updated to add a `value_from_object` override that properly coerces the value to a 32 digit hex string.